### PR TITLE
Make the coordinated shutdown jvm hook work

### LINKED ISSUE #22253: The coordinated shutdown JVM hook does not work
Because we changed `addJvmShutdownHook` to take a code block for Java compatibility, but forgot to change the lambda we pass in into a code block.

### PR DESCRIPTION
**Pull request by johanandren**
Fixes #22253

### CODE REVIEW

**Comment by ktoso** [comment]
OK TO TEST

**Comment by ktoso** [comment]
PLS WHITELIST

**Review by ktoso — APPROVED** [review]
LGTM

File: akka-actor/src/main/scala/akka/actor/CoordinatedShutdown.scala [hunk 1]
```diff
@@ -100,7 +100,7 @@ object CoordinatedShutdown extends ExtensionId[CoordinatedShutdown] with Extensi
   private def initJvmHook(system: ActorSystem, conf: Config, coord: CoordinatedShutdown): Unit = {
     val runByJvmShutdownHook = conf.getBoolean("run-by-jvm-shutdown-hook")
     if (runByJvmShutdownHook) {
-      coord.addJvmShutdownHook { () ⇒
+      coord.addJvmShutdownHook {
```
Comment (patriknw) [inline akka-actor/src/main/scala/akka/actor/CoordinatedShutdown.scala]: ouch

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**